spec/integration/ai_chat_integration_spec.rb

```diff
@@ -0,0 +1,303 @@
+
+      chat.schema = schema_json
+      chat.user("Give me a random number")
+
+      response = chat.generate!
+
+      expect(response).to be_a(Hash)
+      expect(response[:value]).to be_a(Integer)
+      expect(response[:value]).to be_between(1, 10)
+    end
+  end
+
+  describe "previous_response_id functionality" do
+    it "continues a conversation using previous_response_id" do
+      # First conversation
+      chat1 = AI::Chat.new
+      chat1.user("My name is Alice and I live in Boston.")
+      chat1.generate!
+
+      response_id = chat1.previous_response_id
+      expect(response_id).to match(/^resp_/)
+
+      # New conversation continuing from previous
+      chat2 = AI::Chat.new
+      chat2.previous_response_id = response_id
+      chat2.user("What is my name and where do I live?")
+
+      response = chat2.generate!
+
+      expect(response).to match(/Alice/i)
+      expect(response).to match(/Boston/i)
+    end
+
+    it "automatically updates previous_response_id after each generate!" do
+      chat = AI::Chat.new
+
+      expect(chat.previous_response_id).to be_nil
+
+      chat.user("Hello")
+      chat.generate!
+
+      first_id = chat.previous_response_id
+      expect(first_id).to match(/^resp_/)
+
+      chat.user("Goodbye")
+      chat.generate!
+
+      second_id = chat.previous_response_id
+      expect(second_id).to match(/^resp_/)
+      expect(second_id).not_to eq(first_id)
+    end
+  end
+
+  describe "model selection" do
+    it "uses gpt-4.1-nano by default" do
+      chat = AI::Chat.new
+      expect(chat.model).to eq("gpt-4.1-nano")
+
+      chat.user("Hi")
+      chat.generate!
+
+      expect(chat.last[:response].model).to match(/gpt-4/)
+    end
+
+    it "allows setting a different model" do
+      chat = AI::Chat.new
+      chat.model = "gpt-4o-mini"
+
+      chat.user("Hi")
+      chat.generate!
+
+      expect(chat.last[:response].model).to match(/gpt-4o-mini/)
+    end
+  end
+
+  describe "web search functionality" do
+    it "can use web search when enabled" do
+      chat = AI::Chat.new
+      chat.model = "gpt-4o-mini"  # Use a model that supports web search
+      chat.web_search = true
+
+      chat.user("What is the current price of Bitcoin in USD? Please search for the latest information.")
+      response = chat.generate!
+
+      expect(response).to be_a(String)
+      expect(response).to match(/\$|USD|dollar/i)
+    end
+  end
+
+  describe "reasoning effort" do
+    it "accepts reasoning effort levels for o-series models" do
+      chat = AI::Chat.new
+      chat.model = "o1-mini"
+      chat.reasoning_effort = "low"
+
+      expect { chat.reasoning_effort = "low" }.not_to raise_error
+      expect { chat.reasoning_effort = "medium" }.not_to raise_error
+      expect { chat.reasoning_effort = "high" }.not_to raise_error
+      expect { chat.reasoning_effort = :high }.not_to raise_error
```

Comment:
We can also test the other valid Symbol values

```suggestion
      expect { chat.reasoning_effort = "low" }.not_to raise_error
      expect { chat.reasoning_effort = "medium" }.not_to raise_error
      expect { chat.reasoning_effort = "high" }.not_to raise_error
      expect { chat.reasoning_effort = :high }.not_to raise_error
      expect { chat.reasoning_effort = :medium}.not_to raise_error
      expect { chat.reasoning_effort = :low }.not_to raise_error
```